iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/BusAdapter.java

```diff
@@ -214,7 +238,13 @@ public Component getConnectedComponent() {
         ConnectedComponentsManager ccm = getIndex().getView().getConnectedComponentsManager();
         ccm.update();
 
-        return terminals.isEmpty() ? null : ccm.getComponent(terminals.get(0).getConnectedComponentNumber());
+        if (terminals.isEmpty()) {
+            if (terminalRef == null) {
+                return null;
+            }
+            return ccm.getComponent(terminalRef.getConnectedComponentNumber());
+        }
+        return ccm.getComponent(terminals.get(0).getConnectedComponentNumber());
```

Comment:
Why don't we have the same implementation than in `CalculatedBusImpl`?

Reply:
You're right, I think I made it complicated while trying out solutions... Changing it now.